main.go

```diff
@@ -17,84 +17,138 @@ limitations under the License.
+	rw.Write(info)
 }
 
 func EnvHandler(rw http.ResponseWriter, req *http.Request) {
-    environment := make(map[string]string)
-    for _, item := range os.Environ() {
-        splits := strings.Split(item, "=")
-        key := splits[0]
-        val := strings.Join(splits[1:], "=")
-        environment[key] = val
-    }
-
-    envJSON := HandleError(json.MarshalIndent(environment, "", "  ")).([]byte)
-    rw.Write(envJSON)
+	environment := make(map[string]string)
+	for _, item := range os.Environ() {
+		splits := strings.Split(item, "=")
+		key := splits[0]
+		val := strings.Join(splits[1:], "=")
+		environment[key] = val
+	}
+
+	envJSON := HandleError(json.MarshalIndent(environment, "", "  ")).([]byte)
+	rw.Write(envJSON)
 }
 
 func HandleError(result interface{}, err error) (r interface{}) {
-    if err != nil {
-        panic(err)
-    }
-    return result
+	if err != nil {
+		panic(err)
+	}
+	return result
+}
+
+// NrMiddleware is a quick and dirty negroni middleware for transactions
+//  HACK! uses global nrapp....
+// use:
+//   n := negroni.Classic()
+//   n.Use(negroni.HandlerFunc(NrMiddleware)) // <= Important part here
+//   n.UseHandler(r)  // Creating handler outside of scope
+//   n.Run(cfg.serverAddr)
+func NrMiddleware(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	if nrapp != nil {
+		trname := "Unknown"
+
+		// Probably bad, as it contains the IDs...
+		if r != nil && r.URL != nil {
+			trname = r.URL.Path
+		}
+
+		trx := nrapp.StartTransaction(trname, rw, r)
+		defer trx.End()
+	}
+
+	next(rw, r)
 }
 
 func main() {
-    config := newrelic.NewConfig("GuestBook-Go-NewRelic", "New-Relic-Key")
-    app, err := newrelic.NewApplication(config)
-    if err != nil {
-        log.Fatalf("Unable to create NR connection: %v", err)
-    }
-
-    masterPool = simpleredis.NewConnectionPoolHost("redis-master:6379")
-    defer masterPool.Close()
-    slavePool = simpleredis.NewConnectionPoolHost("redis-slave:6379")
-    defer slavePool.Close()
-
-    r := mux.NewRouter()
-    r.HandleFunc(newrelic.WrapHandleFunc(app, "/lrange/{key}", ListRangeHandler)).Methods("GET")
-    r.HandleFunc(newrelic.WrapHandleFunc(app, "/rpush/{key}/{value}", ListPushHandler)).Methods("GET")
-    r.HandleFunc(newrelic.WrapHandleFunc(app, "/info", InfoHandler)).Methods("GET")
-    r.HandleFunc(newrelic.WrapHandleFunc(app, "/env", EnvHandler)).Methods("GET")
-
-    n := negroni.Classic()
-    n.UseHandler(r)
-    n.Run(":3000")
+	var err error
+	var cfg config
+
+	// Parse cli options: https://golang.org/pkg/flag/
+	flag.StringVar(&cfg.serverAddr, "server", ":3000", "Address for server to listen on")
+	flag.StringVar(&cfg.nrLicenseKey, "nrkey", "", "New Relic License Key")
+	flag.StringVar(&cfg.redisMaster, "redis", "localhost:6379", "Redis Master")
+	flag.StringVar(&cfg.redisSlave, "redisslave", "localhost:6379", "Redis Slave")
+	flag.BoolVar(&cfg.debug, "debug", false, "Debug mode")
+	flag.Parse()
```

Comment:
All of this above means you need to use cli args when starting, but don't have to hard-code your stuff. Run like this:
`main -redis "redis-server:6379" -redisslave "redis-slave:6379" -nrkey <KEY>`